src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs

```diff
@@ -660,6 +660,191 @@ void verify(MetadataReference comp1Ref)
             }
         }
 
+        [Fact]
+        public void LanguageVersion_03_CallSite_DelegateNaturalType()
+        {
+            var src1 = @"
+public class Params
+{
+    static public void Test1(params System.ReadOnlySpan<long> a) {}
+    static public void Test2(params long[] a) {}
+}
+";
+            var src2 = @"
+class Program
+{
+    void Test()
+    {
+        var x1 = Params.Test1;
+        var x2 = Params.Test2;
+
+        x1(1);
+        x2(2);
+
+        x1();
+        x2();
+
+        x1([1]);
+        x2([2]);
+    }
+}
+";
+            var comp = CreateCompilation(src2 + src1, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            comp = CreateCompilation(src2 + src1, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.RegularNext);
+            comp.VerifyDiagnostics();
+
+            comp = CreateCompilation(src2 + src1, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseDll, parseOptions: TestOptions.Regular12);
+            comp.VerifyDiagnostics(
+                // (9,9): error CS8652: The feature 'params collections' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         x1(1);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "x1(1)").WithArguments("params collections").WithLocation(9, 9),
```

Comment:
This is honestly not the behavior I expected to see. I was expecting to see the warning on the delegate conversion. Please add a test that looks something like this:

```cs
// Comp 1 (LangVersion Next)
public class Params
{
    static public void Test1(params System.ReadOnlySpan<long> a) {}
    static public void M<T>(T t) {}
}
// Comp 2 (LangVersion 12)
Params.M(Params.Test1); // I expect to see a langversion warning here.
```

Reply:
> Please add a test that looks something like this:

There is no warning for this scenario. If you expect a warning in this case, you need to provide a reasonable rational behind this expectation. From my perspective, the code doesn't take advantage of `params` feature. Remove `params` modifier and nothing fails. Therefore, there is nothing to report about. The language version diagnostics is about preventing breaks when compiling the same code with an older compiler. There is no break like that for this scenario.

Reply:
The type parameter inferred has `params` applied. We synthesize a delegate type in the compilation that has `params` applied, and this is observable through reflection without invoking the delegate, for example. If you remove the `params`, sure, nothing fails, but the code also doesn't mean the same thing.

Reply:
After some offline discussion, here's a more complete example where overload resolution could change:

```cs
// Comp 1
public class Params
{
    static public void Test1(params long[] a) {}
    static public void M<T>(T t) {}
    static public void M(Action<long[]> t) {}
}

// Comp 2
var a = Params.Test1; // A C# 12 compiler would see this as Action<long[]>, a C# 13 compiler would see this as an anonymous delegate
Params.M(a); // A C# 12 compiler will call the Action overload, a C# 13 compiler will call the generic overload.
```